app/controllers/decidim/collaborations/user_collaborations_controller.rb

```diff
@@ -33,7 +33,7 @@ def validate
         reject_user_collaboration if params[:result] == 'ko'
 
         redirect_to EngineRouter.main_proxy(current_feature)
-                    .collaboration_path(collaboration)
+                    .collaboration_path(user_collaboration.collaboration)
```

Comment:
Why this?

Reply:
Mainly to fix some broken specs (confirm_collaboration_spec.rb, I think). The underlying reason (I think) is that the `collaboration` method load the collaboration from the id in the request path, but I don't think the id in this case means the collaboration id but the user collaboration id. The safest thing to do seemed to redirect to the collaboration belonging to the user_collaboration we just updated, and it fixed the specs. Makes sense?

Reply:
Could be. It [tries to get the `collaboration_id` before the `id`](https://github.com/podemos-info/decidim-module-crowdfundings/blob/0e0a243fe841c36f4c38bb2e290dea6d4899008e/app/controllers/concerns/decidim/collaborations/needs_collaboration.rb#L58), but I don't know if it fails in this case. Let's apply your changes and if there is any issue it will appear again when we put everything together.

Reply:
Good point. So this is why:

```
Routes for Decidim::Collaborations::Engine:
confirm_collaboration_user_collaborations POST /collaborations/:collaboration_id/user_collaborations/confirm(.:format) decidim/collaborations/user_collaborations#confirm
              validate_user_collaboration GET  /user_collaborations/:id/validate(.:format)                             decidim/collaborations/user_collaborations#validate
        collaboration_user_collaborations POST /collaborations/:collaboration_id/user_collaborations(.:format)         decidim/collaborations/user_collaborations#create
                           collaborations GET  /collaborations(.:format)                                               decidim/collaborations/collaborations#index
                            collaboration GET  /collaborations/:id(.:format)                                           decidim/collaborations/collaborations#show
                                     root GET  /                                                                       decidim/collaborations/collaborations#index
```

The `valid_user_collaboration` route is mounted as a shallow routes, not under the parent collaboration. That's why the `collaboration_id` is not available.